app/models/laboratory.rb

```diff
@@ -2,9 +2,21 @@
 
 # Laboratory: represents a lab result
 class Laboratory < ApplicationRecord
+  include ValidationHelper
   belongs_to :patient
 
-  validates :result, inclusion: { in: ['positive', 'negative', 'indeterminate', 'other', nil, ''] }
+  %i[result
+     lab_type].each do |enum_field|
+    validates enum_field, inclusion: {
+      in: VALID_PATIENT_ENUMS[enum_field],
+      message: "is not an acceptable value, acceptable values are: '#{VALID_PATIENT_ENUMS[enum_field].reject(&:blank?).join("', '")}'"
+    }
+  end
+
+  %i[specimen_collection
+     report].each do |date_field|
+    validates date_field, date: true
```

Comment:
Is the `date: true` here what triggers the use of the `DateValidator`?

Reply:
Yeah, Rails does some magic and finds the validator which corresponds to `date`. This is the same as how the validation works for emails and phone numbers. If the validator is in a file called `foo_bar_validator`, then 
```
validates some_field, foo_bar: true
```
calls that validator.